src/stream/src/executor/aggregation/agg_group.rs

```diff
@@ -231,81 +286,24 @@ impl<S: StateStore, Strtg: Strategy, const EOWC: bool> AggGroup<S, Strtg, EOWC>
         }
 
         let mut this = Self {
-            group_key,
+            ctx: Context { group_key },
             states,
-            prev_outputs: None, // will be initialized later
+            prev_inter_states: inter_states,
+            prev_outputs: None, // will be set below
             row_count_index,
             _phantom: PhantomData,
         };
 
-        if encoded_states.is_some() {
-            let (_, outputs, _stats) = this.get_outputs(storages, agg_funcs).await?;
+        if !EOWC && this.prev_inter_states.is_some() {
+            let (outputs, _stats) = this.get_outputs(storages, agg_funcs).await?;
             this.prev_outputs = Some(outputs);
         }
 
         Ok(this)
     }
 
-    /// Create a group from encoded states for EOWC. The previous output is set to `None`.
-    #[allow(clippy::too_many_arguments)]
-    pub fn create_eowc(
-        version: PbAggNodeVersion,
-        group_key: Option<GroupKey>,
-        agg_calls: &[AggCall],
-        agg_funcs: &[BoxedAggregateFunction],
-        storages: &[AggStateStorage<S>],
-        encoded_states: &OwnedRow,
-        pk_indices: &PkIndices,
-        row_count_index: usize,
-        extreme_cache_size: usize,
-        input_schema: &Schema,
-    ) -> StreamExecutorResult<Self> {
```

Comment:
Moved to `impl<S: StateStore, Strtg: Strategy> AggGroup<S, Strtg, /* EOWC */ true>` block to avoid misuse.